zlib/grabber.go

```diff
@@ -315,14 +313,21 @@ func makeHTTPGrabber(config *Config, grabData *GrabData) func(string, string, st
 			httpHost = hostWithoutPort
 		}
 
+		var req *http.Request
+		var resp *http.Response
+
 		switch config.HTTP.Method {
 		case "GET":
-			resp, err = client.GetWithHost(fullURL, httpHost)
+			req, err = http.NewRequestWithHost("GET", fullURL, httpHost, nil)
```

Comment:
Can you explain why the call changes here?

Reply:
Sure -- the Get/HeadWithHost does the request immediately, before we get a chance to add any headers. It would be nice if you could set default headers in the client, but I don't see an API for that. The docs for GetWithHost say "To make a request with custom headers, use NewRequest and Client.Do."

Reply:
Got it! Thanks for the explanation!